### ISSUE TITLE
PEP 541 Request: pyutilz collides with... nothing?

### ISSUE THREAD

**Issue by scottgasch** [pypa/pypi-support#2201]
### Project to be claimed

`pyutilz`: https://pypi.org/project/pyutilz

### Your PyPI username

`sgasch`: https://pypi.org/user/sgasch

### Reasons for the request

I'm trying to upload a new project with the name pyutilz and am getting an error:

    400 The name 'pyutilz' is too similar to an existing project. See https://pypi.org/help/#project-name for more information.

I don't see any existing project using the same name.  I do see many other projects with names similar to py[thon]-util[itie[s|z]].  Given that we're using a flat namespace for projects, I thought that finding an unclaimed name would be sufficient to upload.  If not, can you explain the logic used to detect names that are "too similar" to others?  Is it hamming
distance?  What is acceptable?

I've been testing with this name on https://test.pypi.org/project/pyutilz/ for a couple of days and was surprised by this
error on trying to upload to PyPi (non-test).

Please advise about similarity requirements and consider suggesting a name.

Thx,
Scott

### Source code repositories URLs

https://test.pypi.org/project/pyutilz/

### Contact and additional research

None

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by scottgasch** [comment]
I've pinged the owner of pyutils which seems abandoned and is likely the project this is too similarly named to.  No response yet.  It's 1 version and 500 bytes in size... we'll see if they get back to me.

**Comment by scottgasch** [comment]
The owner of pyutils responded saying it was an old project and he deleted it.  I still can't upload mine because it says the name it too similar.  Is there a delay in the index?  Should I change my project name to match his (pyutilz -> pyutils)?  Please advise.

**Comment by scottgasch** [comment]
Well, pyutils seems to be deleted now.  As I mentioned in the previous comment, I tried to upload pyutilz again and it failed.  I also changed the project name to pyutils and that, too, fails with another "too similar" message.  There is a pyutil2 that is perhaps the problem.

Can you please tell me if I'm fighting a losing battle here and should instead name this package something like scotts_python_utilities?

**Comment by scottgasch** [comment]
Hello?

**Comment by yeraydiazdiaz** [comment]
Hello @scottgasch, I believe your project name is clashing with https://pypi.org/project/py-utilz/ which is fairly recent. Have you considered a different name for your project?

**Comment by scottgasch** [comment]
Would it be ok to use pyutils, which is the name of the project I originally suspected was causing problems?  It was stale for ~12 years and, when I talked to the owner via email, he deleted the project.  I've tried to use it, actually, and had the same problem (pypi claims the name is too similar to an existing project).

**Comment by yeraydiazdiaz** [comment]
Yes, you run into the same similarity problem since there are other similarly named projects. The only thing I can think of if you really like the name is to open a separate PEP 541 request for https://pypi.org/project/py_utils/ which has no functionality.

**Comment by scottgasch** [comment]
So, there is no pyutilz right now and I got pyutils to delete itself so it
also no longer exists... but I can't use either of these.  You'd rather me
create a new PEP541 request to take over py_utils which does exist?

**Comment by yeraydiazdiaz** [comment]
I'm simply showing you the options that are available at this time in PyPI. You are the one insisting on using a quite generic project name that has had many permutations of the same since many year ago.

**Comment by scottgasch** [comment]
I don't understand how is it ok to attempt to use py_utils which is an
existing project but not to use pyutilz or pyutils when neither of those
projects exist?  Should I have not gotten the pyutils guy to delete his
project and just tried to take it over instead?  Why is py_utils an ok
project name while pyutilz isn't?  Is it because it exists already?

I will think about another name if this is too generic but I really would
not like one that is super long and it seems PyPI's rules make that
basically unavoidable at this point IIUC.

**Comment by di** [comment]
I've transferred `pyutils` to `sgasch`, you should be able to upload now.

**Comment by scottgasch** [comment]
Thanks!
